docs/cookbook/makeExpandedTemporal.mjs

```diff
@@ -0,0 +1,173 @@
+function bigIntAbs(n) {
+  if (n < 0n) return -n;
+  return n;
+}
+
+// The years are unlimited, but for output purposes we assume 10 digits,
+// because ISO 8601 requires the expanded year format to pick a consistent
+// number of digits.
+function formatExpandedYear(year) {
+  let yearString;
+  if (year < 1000 || year > 9999) {
+    let sign = year < 0 ? '-' : '+';
+    let yearNumber = bigIntAbs(year);
+    yearString = sign + `${yearNumber}`.padStart(10, '0');
+  } else {
+    yearString = `${year}`;
+  }
+  return yearString;
+}
+
+function isLeapYear(year) {
+  const isDiv4 = year % 4n === 0n;
+  const isDiv100 = year % 100n === 0n;
+  const isDiv400 = year % 400n === 0n;
+  return isDiv4 && (!isDiv100 || isDiv400);
+}
+
+// This checks to see if the ISO string matches our 10-digit expanded year
+// format, and if so, returns both the expanded year as a BigInt, and a new
+// ISO string with an in-range year that can be passed to the original
+// Temporal string parsing functions.
+// The in-range year is 1972 if the expanded year is a leap year, and
+// otherwise 1970, so that the rules for February 29 remain correct.
+// See the note about the number of digits in formatExpandedYear().
+function parseExpandedYear(isoString) {
+  const matchExpandedYear = /^[-+\u2122]\d{10}/;
```

Comment:
I think I'm missing something, is `\u2122` meant to be here? Isn't that `'™'`? Even without the unicode flag set, I'm a little confused.

I mean, I expect trademark law to outlive us all, but I imagine I must be missing something here :stuck_out_tongue:

Reply:
It was meant to be U+2212 (minus sign)

Reply:
Haha excellent, unicode typos are a class of bugs I hadn't thought about. A very endearing bug.

Reply:
> A very endearing bug.

Endearing Bug™ Indeed!